src/AzureClient/AzureClient.cs

```diff
@@ -72,6 +82,37 @@ public class AzureClient : IAzureClient
             string workspaceName,
             string storageAccountConnectionString,
             bool refreshCredentials = false)
+        {
+            var connectionResult = await ConnectToWorkspaceAsync(channel, subscriptionId, resourceGroupName, workspaceName, refreshCredentials);
+            if (connectionResult.Status != ExecuteStatus.Ok)
+            {
+                return connectionResult;
+            }
+
+            if (ActiveWorkspace == null)
+            {
+                return AzureClientError.WorkspaceNotFound.ToExecutionResult();
```

Comment:
Is this something that we would like to log as an error?

Reply:
Returning an error here causes the magic command `%azure.connect` to return an error status, which will be logged and communicated back to the user.

(This particular case should never really happen, since the current implementation of `ConnectToWorkspaceAsync` should have always set `ActiveWorkspace` if it returns `ExecuteStatus.Ok`, but it seems necessary to keep this here to convince the compiler that `ActiveWorkspace` truly is not `null` when using it below.)